ocp_resources/resource.py

```diff
@@ -101,15 +101,22 @@ def get_client(
     Returns:
         DynamicClient: a kubernetes client.
     """
+    client_configuration = kwargs.get("client_configuration", kubernetes.client.Configuration())
+
+    # If the proxy is not set, set it from the environment
+    if not client_configuration.proxy:
+        proxy = os.environ.get("HTTPS_PROXY") or os.environ.get("HTTP_PROXY")
```

Comment:
As far as I know, they don't have to match. So you need to take each ENV individually.

Reply:
right

Reply:
```
if configuration.proxy and not should_bypass_proxies(configuration.host, no_proxy=configuration.no_proxy or ''):
            self.pool_manager = urllib3.ProxyManager(
                num_pools=pools_size,
                maxsize=maxsize,
                cert_reqs=cert_reqs,
                ca_certs=ca_certs,
                cert_file=configuration.cert_file,
                key_file=configuration.key_file,
                proxy_url=configuration.proxy,
                proxy_headers=configuration.proxy_headers,
                **addition_pool_args
            )
```

This is what pass to `urllib3` > `proxy_url`